src/components/views/emojipicker/Header.tsx

```diff
@@ -36,10 +36,10 @@ class Header extends React.PureComponent<IProps> {
         let current: number;
         // As multiple categories may be visible at once, we want to find the one closest to the relative direction
         if (delta < 0) {
-            current = this.props.categories.findIndex((c) => c.visible);
+            current = this.props.categories.findIndex((c) => c.firstVisible);
         } else {
             // XXX: Switch to Array::findLastIndex once we enable ES2023
-            current = findLastIndex(this.props.categories, (c) => c.visible);
+            current = findLastIndex(this.props.categories, (c) => c.firstVisible);
```

Comment:
Could we not just leverage `findIndex`, given categories are a total order and only categories next to each other may be visible together we could just use `findIndex`/`findLastIndex` to get first/last as necessary, and then not need to store `firstVisible`.

In the for-each loop in EmojiPicker we could just store `let firstVisible: Category` calculated in the loop